codex-cli/scripts/run_in_container.sh

```diff
@@ -51,14 +68,21 @@ cleanup
 # Run the container with the specified directory mounted at the same path inside the container.
 docker run --name "$CONTAINER_NAME" -d \
   -e OPENAI_API_KEY \
+  -e ALLOWED_DOMAINS="$ALLOWED_DOMAINS" \
   --cap-add=NET_ADMIN \
   --cap-add=NET_RAW \
   -v "$WORK_DIR:/app$WORK_DIR" \
   codex \
   sleep infinity
 
+# Convert space-separated domains to array of arguments for init_firewall.sh
+domain_args=()
+for domain in $ALLOWED_DOMAINS; do
+  domain_args+=("$domain")
+done
+
 # Initialize the firewall inside the container.
-docker exec "$CONTAINER_NAME" bash -c "sudo /usr/local/bin/init_firewall.sh"
+docker exec "$CONTAINER_NAME" bash -c "sudo /usr/local/bin/init_firewall.sh ${domain_args[*]}"
```

Comment:
This makes me slightly uncomfortable because it feels like a malicious input to `--allowed_domains` could run arbitrary code in your container (e.g., `' && rm -rf /'`), though I suppose if an attacker can manipulate `--allowed_domains`, it's already game over?

Reply:
Perhaps what we should be doing here is not adding this script to `/etc/sudoers.d/node-firewall` over in `Dockerfile` and instead do this via `docker exec --user root` so that the `node` user inside the container cannot run this script?

Reply:
addressed. thx!